### PR TITLE
[GOBBLIN-2145] fix bug in getting flow status

### PR DESCRIPTION
**Pull request by arjun4084346**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-2145

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
method `getAllFlowStatusesForFlowExecutionsOrdered` expects flowGroup, flowName in this order, while it was being passed the parameters in opposite order.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
updated tests

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-runtime/src/main/java/org/apache/gobblin/service/monitoring/JobStatusRetriever.java [hunk 1]
```diff
@@ -87,25 +87,14 @@ public abstract Iterator<JobStatus> getJobStatusesForFlowExecution(String flowNa
    * is decided by comparing {@link JobStatus#getFlowExecutionId()}.
    * @return `FlowStatus`es are ordered by descending flowExecutionId.
    **/
-  public abstract List<FlowStatus> getAllFlowStatusesForFlowExecutionsOrdered(String flowGroup,String flowName);
+  public abstract List<FlowStatus> getAllFlowStatusesForFlowExecutionsOrdered(String flowGroup, String flowName);
 
 
   public long getLatestExecutionIdForFlow(String flowName, String flowGroup) {
```
Comment (phet) [inline gobblin-runtime/src/main/java/org/apache/gobblin/service/monitoring/JobStatusRetriever.java:93]: this one is still in the reverse order

File: gobblin-runtime/src/test/java/org/apache/gobblin/service/monitoring/FlowStatusGeneratorTest.java [hunk 2]
```diff
@@ -58,12 +58,12 @@ public void testIsFlowRunningCompiledPastExecution() {
     JobStatus jobStatus = JobStatus.builder().flowGroup(flowGroup).flowName(flowName).flowExecutionId(flowExecutionId)
         .jobName(JobStatusRetriever.NA_KEY).jobGroup(JobStatusRetriever.NA_KEY).eventName(ExecutionStatus.COMPILED.name()).build();
     Iterator<JobStatus> jobStatusIterator = Lists.newArrayList(jobStatus).iterator();
-    FlowStatus flowStatus = new FlowStatus(flowName,flowGroup,flowExecutionId,jobStatusIterator,ExecutionStatus.COMPILED);
-    when(jobStatusRetriever.getAllFlowStatusesForFlowExecutionsOrdered(flowName, flowGroup)).thenReturn(
+    FlowStatus flowStatus = new FlowStatus(flowName, flowGroup, flowExecutionId, jobStatusIterator, ExecutionStatus.COMPILED);
+    when(jobStatusRetriever.getAllFlowStatusesForFlowExecutionsOrdered(flowName, flowName)).thenReturn(
```
Comment (phet) [inline gobblin-runtime/src/test/java/org/apache/gobblin/service/monitoring/FlowStatusGeneratorTest.java:62]: dup of `flowName`

File: gobblin-runtime/src/test/java/org/apache/gobblin/service/monitoring/FlowStatusGeneratorTest.java [hunk 3]
```diff
@@ -77,12 +77,12 @@ public void skipFlowConcurrentCheckSameFlowExecutionId() {
     JobStatus jobStatus = JobStatus.builder().flowGroup(flowGroup).flowName(flowName).flowExecutionId(flowExecutionId)
         .jobName(JobStatusRetriever.NA_KEY).jobGroup(JobStatusRetriever.NA_KEY).eventName(ExecutionStatus.COMPILED.name()).build();
     Iterator<JobStatus> jobStatusIterator = Lists.newArrayList(jobStatus).iterator();
-    FlowStatus flowStatus = new FlowStatus(flowName,flowGroup,flowExecutionId,jobStatusIterator,ExecutionStatus.COMPILED);
-    when(jobStatusRetriever.getAllFlowStatusesForFlowExecutionsOrdered(flowName, flowGroup)).thenReturn(
+    FlowStatus flowStatus = new FlowStatus(flowName, flowGroup, flowExecutionId, jobStatusIterator,ExecutionStatus.COMPILED);
```
Comment (phet) [inline gobblin-runtime/src/test/java/org/apache/gobblin/service/monitoring/FlowStatusGeneratorTest.java:80]: it's confusing that `org.apache.gobblin.service.monitoring.FlowStatus` uses the (name, group) ordering.  personally, I SO MUCH prefer (group, name), but the fact that some of these other long-standing classes don't follow it, makes me question whether we ought to just retreat and accept the (name, group) ordering as a fact of life.

if we want to double down on (group, name) then I suggest to change it EVERYWHERE, even for `FlowStatus`

File: gobblin-service/src/test/java/org/apache/gobblin/service/monitoring/MysqlJobStatusRetrieverTest.java [hunk 4]
```diff
@@ -97,31 +97,31 @@ public void testGetFlowStatusFromJobStatuses() throws Exception {
 
     addJobStatusToStateStore(flowExecutionId, JobStatusRetriever.NA_KEY, ExecutionStatus.COMPILED.name());
     Assert.assertEquals(ExecutionStatus.COMPILED,
-        jobStatusRetriever.getFlowStatusFromJobStatuses(jobStatusRetriever.getJobStatusesForFlowExecution(FLOW_NAME, FLOW_GROUP, flowExecutionId)));
+        JobStatusRetriever.getFlowStatusFromJobStatuses(jobStatusRetriever.getJobStatusesForFlowExecution(FLOW_NAME, FLOW_GROUP, flowExecutionId)));
```
Comment (phet) [inline gobblin-service/src/test/java/org/apache/gobblin/service/monitoring/MysqlJobStatusRetrieverTest.java:100]: again, let's reorder only if we go ALL-IN w/ this `getJobStatusesForFlowExecution` too

**Review by phet — COMMENTED** [review]
I'm referring to these in `JobStatusRetriever`:
```
  public abstract Iterator<JobStatus> getJobStatusesForFlowExecution(String flowName, String flowGroup,
      long flowExecutionId);
  public abstract Iterator<JobStatus> getJobStatusesForFlowExecution(String flowName, String flowGroup,
      long flowExecutionId, String jobName, String jobGroup);
```

File: gobblin-runtime/src/test/java/org/apache/gobblin/service/monitoring/FlowStatusGeneratorTest.java (hunk 3 above)
Comment (arjun4084346) [reply gobblin-runtime/src/test/java/org/apache/gobblin/service/monitoring/FlowStatusGeneratorTest.java:80]: i agree many old methods have flowName first, but a lot of methods also have flowGroup first.

Comment (arjun4084346) [reply gobblin-runtime/src/test/java/org/apache/gobblin/service/monitoring/FlowStatusGeneratorTest.java]: yea, i am not comfortable with this state either. lets change it in some other PR, this one is quite urgent

**Review by Blazer-007 — APPROVED** [review]

**Review by phet — COMMENTED** [review]
I see what you did to fix the mis-ordering introduced here - https://github.com/apache/gobblin/pull/3989/files#diff-237d0e6a4a38c77a7a26a49075a7d8dfa5b95002adff25a18586f7495dca9820R157

but really that new method `JobStatusRetriever::getAllFlowStatusesForFlowExecutionsOrdered` is the only one of its class w/ the (flowGroup, flowName) param ordering.

I understand it turned into too much effort to reorder all the rest to (flowGroup, flowName).  the converse of reordering only that one new method to use (flowName, flowGroup), as the other legacy ones do, might arguably leave the codebase more consistent.

**Pull request merged**